### PR TITLE
added float serializer implementation and test

### PR DESCRIPTION
**Pull request by NNRepos**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

* related to #138

### CODE REVIEW

**Comment by sanastas** [comment]
Thank you for your contribution! We will review it on Wednesday.

**Comment by NNRepos** [comment]
@sanastas, any updates?

File: core/src/test/java/com/yahoo/oak/BuildAllTypes.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    public static int MEBIBYTE = 1024 * 1024;
+
+    
+    @Test
+    public void testBuildIntFloat(){
+        OakIntSerializer int_serializer = new OakIntSerializer();
+        OakFloatSerializer float_serializer = new OakFloatSerializer();
+        OakIntComparator int_comparator = new OakIntComparator();
+
+        OakMapBuilder<Integer, Float> builder = new OakMapBuilder<Integer, Float>(int_comparator, int_serializer, float_serializer, Integer.MIN_VALUE)
```
Comment (sanastas) [inline core/src/test/java/com/yahoo/oak/BuildAllTypes.java:23]: Please make the line shorter

File: core/src/test/java/com/yahoo/oak/BuildAllTypes.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+            .setMemoryCapacity(MEBIBYTE); // 1MB in bytes
+
+    OakMap<Integer,Float> oak = builder.build();
+    
+    int my_key = 0;
+    float my_val = (float) 3.14;
+    oak.put(my_key, my_val);
+
+    Assert.assertTrue(oak.get(my_key)==my_val);
```
Comment (sanastas) [inline core/src/test/java/com/yahoo/oak/BuildAllTypes.java:32]: Please add a check for new float comparator as well

File: core/src/test/java/com/yahoo/oak/BuildAllTypes.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+
+    }
+
+    // TODO add all type combinations
```
Comment (sanastas) [inline core/src/test/java/com/yahoo/oak/BuildAllTypes.java:37]: Is it possible to add some other types as well? Maybe just part of them...

**Review by sanastas — CHANGES_REQUESTED** [review]
Thank you! Looks good to me, small comments inside.

**Comment by sanastas** [comment]
> @sanastas, any updates?

@NNRepos thank you! I have taken a look and it looks good. I left some small comments, please let me know what do you think. I also assume that all Oak tests pass successfully, right?

**Comment by NNRepos** [comment]
@sanastas done. is this ok?

**Comment by sanastas** [comment]
yes looks good

**Comment by sanastas** [comment]
When I tried `mvn clean package` myself on your code, there are some code styling errors:

[ERROR] /Users/anastas/WorkDesk/DRUIDS/otherOakRepos/Oak/core/src/test/java/com/yahoo/oak/BuildAllTypes.java:1: Line does not match expected header line of '/*'. [Header]
[ERROR] /Users/anastas/WorkDesk/DRUIDS/otherOakRepos/Oak/core/src/test/java/com/yahoo/oak/BuildAllTypes.java:6: Extra separation in import group before 'com.yahoo.oak.common.floatnum.OakFloatComparator' [ImportOrder]
[ERROR] /Users/anastas/WorkDesk/DRUIDS/otherOakRepos/Oak/core/src/test/java/com/yahoo/oak/BuildAllTypes.java:6: Wrong order for 'com.yahoo.oak.common.floatnum.OakFloatComparator' import. [ImportOrder]
[ERROR] /Users/anastas/WorkDesk/DRUIDS/otherOakRepos/Oak/core/src/test/java/com/yahoo/oak/BuildAllTypes.java:9: Extra separation in import group before 'com.yahoo.oak.common.integer.OakIntSerializer' [ImportOrder]
[ERROR] /Users/anastas/WorkDesk/DRUIDS/otherOakRepos/Oak/core/src/test/java/com/yahoo/oak/BuildAllTypes.java:11: Wrong order for 'com.yahoo.oak.common.integer.OakIntComparator' import. [ImportOrder]
[ERROR] /Users/anastas/WorkDesk/DRUIDS/otherOakRepos/Oak/core/src/test/java/com/yahoo/oak/BuildAllTypes.java:15:23: Variable 'MEBIBYTE' must be private and have accessor methods. [VisibilityModifier]
[ERROR] /Users/anastas/WorkDesk/DRUIDS/otherOakRepos/Oak/core/src/test/java/com/yahoo/oak/BuildAllTypes.java:19:26: Name 'int_serializer' must match pattern '^[a-z][a-zA-Z0-9]*$'. [LocalVariableName]
[ERROR] /Users/anastas/WorkDesk/DRUIDS/otherOakRepos/Oak/core/src/test/java/com/yahoo/oak/BuildAllTypes.java:20:26: Name 'int_serializer2' must match pattern '^[a-z][a-zA-Z0-9]*$'. [LocalVariableName]
[ERROR] /Users/anastas/WorkDesk/DRUIDS/otherOakRepos/Oak/core/src/test/java/com/yahoo/oak/BuildAllTypes.java:21:26: Name 'int_comparator' must match pattern '^[a-z][a-zA-Z0-9]*$'. [LocalVariableName]
[ERROR] /Users/anastas/WorkDesk/DRUIDS/otherOakRepos/Oak/core/src/test/java/com/yahoo/oak/BuildAllTypes.java:28:13: Name 'my_key' must match pattern '^[a-z][a-zA-Z0-9]*$'. [LocalVariableName]
[ERROR] /Users/anastas/WorkDesk/DRUIDS/otherOakRepos/Oak/core/src/test/java/com/yahoo/oak/BuildAllTypes.java:29:13: Name 'my_val' must match pattern '^[a-z][a-zA-Z0-9]*$'. [LocalVariableName]
[ERROR] /Users/anastas/WorkDesk/DRUIDS/otherOakRepos/Oak/core/src/test/java/com/yahoo/oak/BuildAllTypes.java:38:26: Name 'int_serializer' must match pattern '^[a-z][a-zA-Z0-9]*$'. [LocalVariableName]
[ERROR] /Users/anastas/WorkDesk/DRUIDS/otherOakRepos/Oak/core/src/test/java/com/yahoo/oak/BuildAllTypes.java:39:28: Name 'float_serializer' must match pattern '^[a-z][a-zA-Z0-9]*$'. [LocalVariableName]
[ERROR] /Users/anastas/WorkDesk/DRUIDS/otherOakRepos/Oak/core/src/test/java/com/yahoo/oak/BuildAllTypes.java:40:26: Name 'int_comparator' must match pattern '^[a-z][a-zA-Z0-9]*$'. [LocalVariableName]
[ERROR] /Users/anastas/WorkDesk/DRUIDS/otherOakRepos/Oak/core/src/test/java/com/yahoo/oak/BuildAllTypes.java:47:13: Name 'my_key' must match pattern '^[a-z][a-zA-Z0-9]*$'. [LocalVariableName]
[ERROR] /Users/anastas/WorkDesk/DRUIDS/otherOakRepos/Oak/core/src/test/java/com/yahoo/oak/BuildAllTypes.java:48:15: Name 'my_val' must match pattern '^[a-z][a-zA-Z0-9]*$'. [LocalVariableName]
[ERROR] /Users/anastas/WorkDesk/DRUIDS/otherOakRepos/Oak/core/src/test/java/com/yahoo/oak/BuildAllTypes.java:57:28: Name 'float_serializer' must match pattern '^[a-z][a-zA-Z0-9]*$'. [LocalVariableName]
[ERROR] /Users/anastas/WorkDesk/DRUIDS/otherOakRepos/Oak/core/src/test/java/com/yahoo/oak/BuildAllTypes.java:58:29: Name 'string_serializer' must match pattern '^[a-z][a-zA-Z0-9]*$'. [LocalVariableName]
[ERROR] /Users/anastas/WorkDesk/DRUIDS/otherOakRepos/Oak/core/src/test/java/com/yahoo/oak/BuildAllTypes.java:59:28: Name 'float_comparator' must match pattern '^[a-z][a-zA-Z0-9]*$'. [LocalVariableName]
[ERROR] /Users/anastas/WorkDesk/DRUIDS/otherOakRepos/Oak/core/src/test/java/com/yahoo/oak/BuildAllTypes.java:66:15: Name 'my_key' must match pattern '^[a-z][a-zA-Z0-9]*$'. [LocalVariableName]
[ERROR] /Users/anastas/WorkDesk/DRUIDS/otherOakRepos/Oak/core/src/test/java/com/yahoo/oak/BuildAllTypes.java:67:16: Name 'my_val' must match pattern '^[a-z][a-zA-Z0-9]*$'. [LocalVariableName]

Can you please fix it? Thank I will merge this PR

**Comment by NNRepos** [comment]
@sanastas alright.
i've reformatted the file - both the tests and `mvn clean` worked for me

**Comment by sanastas** [comment]
@NNRepos ,

Thank you! Still when I do `mvn clean package` I get one last check style error.

`[ERROR] /Users/anastas/WorkDesk/DRUIDS/otherOakRepos/Oak/core/src/test/java/com/yahoo/oak/BuildAllTypes.java:15: Extra separation in import group before 'org.junit.Assert' [ImportOrder]`

It appears to me `mvn clean` shows not the same results as `mvn clean package`. Anyway, please fix that and I will merge this PR.

**Comment by NNRepos** [comment]
@sanastas done. i think my VScode only supports `mvn clean`, so i can't confirm that it worked

**Comment by NNRepos** [comment]
also, i recommend adding a PR check in GitHub which runs `mvn clean package` automatically. I can open a new issue if you're interested.

**Comment by sanastas** [comment]
> @sanastas done. i think my VScode only supports mvn clean, so i can't confirm that it worked

@NNRepos , thank you! I verified that all code style checks pass. However, I payed attention that your test didn't run automatically and this is because all test files must have names: *Test.java Please fix the file and class name. I did the change locally and saw that all is good after that fix.

> also, i recommend adding a PR check in GitHub which runs mvn clean package automatically. I can open a new issue if you're interested.

Of course, go ahead and open an issue. If you could add the PR check in GitHub by yourself it could be wonderful. Otherwise, please explain in the issue how to do it. Because I do not know :)

**Comment by NNRepos** [comment]
@sanastas will do

**Comment by sanastas** [comment]
@NNRepos ,

Congratulations! Your PR was the first to be merged as part of the Yahoo Hackathon! Thank you for your contribution! It would be great to continue collaboration!

Are you from Verizon Media?

**Comment by NNRepos** [comment]
@sanastas thanks!
no, i'm from the technion.

**Comment by sanastas** [comment]
@NNRepos , that is interesting, I have done all my degrees in Technion and lectured there many semesters including the last winter semester.

Would you be interested in any future collaboration? Is there any way to contact you other than writing a comment here?

**Comment by NNRepos** [comment]
@sanastas not really, as i'm already employed 😅 
thanks again :)

**Pull request merged**